### PR TITLE
adapter options

### PR DESCRIPTION
**Pull request by Perfect-Web**
add adapter options for http client

### CODE REVIEW

File: config/apigility-consumer.local.php.dist [hunk 1]
```diff
@@ -13,6 +13,14 @@ return [
             'client_secret' => 'foo_s3cret',
         ],
         
+        // adapter options
+        'api-http-adapter-options' => [
+            'curloptions' => array(
```
Comment (samsonasik) [inline config/apigility-consumer.local.php.dist:18]: short array syntax

File: src/Service/ClientServiceFactory.php [hunk 2]
```diff
@@ -14,7 +14,11 @@ class ClientServiceFactory
     public function __invoke($container)
     {
         $config     = $container->get('config');
-        $httpClient = new HttpClient();
+        $httpClient = new HttpClient(
+            null,
+            !empty($config['apigility-consumer']['api-http-adapter-options']) ?
```
Comment (samsonasik) [inline src/Service/ClientServiceFactory.php:19]: as curl only supposed to be used for digest, it should only happen in `ClientService::callAPI()` using `$this->httpClient->setOptions()`.

**Comment by samsonasik** [comment]
Lastly, please provide unit test. Also, please create new branch instead of master to start pull request https://akrabat.com/the-beginners-guide-to-contributing-to-a-github-project/

File: src/Service/ClientServiceFactory.php (hunk 2 above)
Comment (Perfect-Web) [reply src/Service/ClientServiceFactory.php:19]: you dont need to complicate that much as httpclient only used that in curl adapters, so it is enough to pass it at construct

Comment (samsonasik) [reply src/Service/ClientServiceFactory.php:19]: no, we don't need curl as it only happen in digest http auth, also, please provide unit test

Comment (Perfect-Web) [reply src/Service/ClientServiceFactory.php:19]: its fine, im gonna work on my own fork and continue there, obvious the perspectives divege

**Pull request closed without merging**